src/classes/ADDR_CannotDelete_TDTM.cls

```diff
@@ -0,0 +1,89 @@
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE 
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2020
+* @group Addresses
+* @group-content ../../ApexDocContent/Addresses.htm
+* @description Stops an Address from being deleted if it has any Account or Contact child records. 
+*/
+public with sharing class ADDR_CannotDelete_TDTM extends TDTM_Runnable {
+    /*******************************************************************************************************
+    * @description Get the setting of preventing Address deletion
+    */
+    private static Boolean enabledPreventAddressDeletion = UTIL_CustomSettingsFacade.getSettings().Prevent_Address_Deletion__c;
+    
+    /*******************************************************************************************************
+    * @description Stops an Address from being deleted if it has any Account or Contact child records. 
+    * @param listNew the list of Addresses from trigger new. 
+    * @param listOld the list of Addresses from trigger old. 
+    * @param triggerAction which trigger event (BeforeInsert, AfterInsert, etc.). 
+    * @param objResult the describe for Addresses 
+    * @return dmlWrapper.  
+    ********************************************************************************************************/
+    public override DmlWrapper run(List<SObject> newlist, List<SObject> oldlist, 
+    TDTM_Runnable.Action triggerAction, Schema.DescribeSObjectResult objResult) {
+        
+        if (!enabledPreventAddressDeletion) {
+            return new DmlWrapper(); 
+        }
+        
+        Map<ID, Address__c> oldmap = new Map<ID, Address__c>((List<Address__c>)oldList);
+        
+        if (triggerAction == TDTM_Runnable.Action.BeforeDelete) {
+            for (Address__c a : [
+                                    SELECT Id, 
+                                        (SELECT Id FROM Address__c.Accounts__r LIMIT 1),
+                                        (SELECT Id FROM Address__c.Contacts1__r LIMIT 1)
+                                    FROM Address__c WHERE ID in :oldlist
+                                    
+                                ]) {
+                
+                if (hasChildRecords(a)) {
+                    Address__c addressInContext = oldMap.get(a.ID);
+                    addressInContext.addError(Label.CannotDelete);
+                }
+            }     
+        }
+
+        return new DmlWrapper();
+    }
+    
+    /*******************************************************************************************************
+     * @description Evaluates whether the Address has any child (Account or Contact) related records.
+     * @param a is the current Address record.
+     * @return Boolean.
+     ********************************************************************************************************/
+    private static Boolean hasChildRecords(Address__c a) {
+        
```

Comment:
Can you please remove static here ref: https://github.com/SalesforceFoundation/EDA/pull/1165#discussion_r416075994, add a unit test method to test this method by adding @testVisible and also change the parameter name from 'a'?